middleware/proxy.go

```diff
@@ -231,7 +237,17 @@ func ProxyWithConfig(config ProxyConfig) echo.MiddlewareFunc {
 
 			req := c.Request()
 			res := c.Response()
-			tgt := config.Balancer.Next(c)
+
+			var tgt *ProxyTarget
+			if isTargetProvider {
+				tgt, err = provider.NextTarget(c)
+				if err != nil {
+					c.String(http.StatusInternalServerError, err.Error())
```

Comment:
Please change this back to `return err` as it was. If `c.String(http.StatusInternalServerError, err.Error())` is used this would mean that after this line the Response to the client has been "commited" and sent to the client and no middleware up in chain can change anything. If `return err` is used then middlewares or global error handler can act on that error.

Reply:
Done. Now I wonder isn't it better the interface to be defined as _NextTarget(echo.Context) (*ProxyTarget, echo.HTTPError)_ because returning simple errors do not return the error message but Internal Server Error